pkg/scheduler/objects/preemption.go

```diff
@@ -349,6 +349,7 @@ func (p *Preemptor) checkPreemptionPredicates(predicateChecks []*si.PreemptionPr
 		result := &predicateCheckResult{
 			allocationKey: check.AllocationKey,
 			nodeID:        check.NodeID,
+			instType:      UnknownInstanceType,
```

Comment:
Let's keep the instance type out of the preemption code. It's only needed on allocations. Once a node is chosen for the allocation we can apply the instance type.

Reply:
As I indicated above: Only the instance type of the node assigned to the NEW allocation matters -- this has nothing to do with the preemption logic. Refactor this so that the instance type is handled outside the preemption code.